### PR TITLE
PP-5403 - Process `parent_resource_external_id` to transaction

### PR DESCRIPTION
**Pull request by kbottla**

### CODE REVIEW

File: src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java [hunk 1]
```diff
@@ -29,19 +29,20 @@
...
     private static final String UPSERT_STRING =
             "INSERT INTO transaction(" +
-                "external_id,gateway_account_id,amount,description,reference,state,email,cardholder_name," +
+                "external_id,parent_transaction_id,gateway_account_id,amount,description,reference,state,email,cardholder_name," +
                 "external_metadata,created_date,transaction_details,event_count,card_brand, " +
                 "last_digits_card_number,first_digits_card_number,net_amount,total_amount,fee,type" +
             ") " +
             "VALUES (" +
-                ":externalId,:gatewayAccountId,:amount,:description,:reference,:state,:email,:cardholderName," +
+                ":externalId,:parentExternalId,:gatewayAccountId,:amount,:description,:reference,:state,:email,:cardholderName," +
```
Comment (sfount) [inline src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java]: Is there a reason both `parent_transaction_id` and `parentExternalId` are used? Agreed that getting external in the name would be a good thing but we should keep it consistent. 

`parent_external_id`
`parent_transaction_external_id`

Comment (kbottla) [reply src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java]: Renamed in https://github.com/alphagov/pay-ledger/pull/147

File: src/main/java/uk/gov/pay/ledger/transaction/entity/TransactionEntity.java [hunk 2]
```diff
@@ -255,6 +267,11 @@ public Builder withExternalId(String externalId) {
             return this;
         }
 
+        public Builder withParentExternalId(String parentTransactionId) {
```
Comment (sfount) [inline src/main/java/uk/gov/pay/ledger/transaction/entity/TransactionEntity.java]:
```suggestion
        public Builder withParentExternalId(String parentExternalId) {
```

File: src/main/java/uk/gov/pay/ledger/transaction/dao/mapper/TransactionMapper.java [hunk 3]
```diff
@@ -19,6 +19,7 @@ public TransactionEntity map(ResultSet rs, StatementContext ctx) throws SQLExcep
                 .withId(rs.getLong("id"))
                 .withGatewayAccountId(rs.getString("gateway_account_id"))
                 .withExternalId(rs.getString("external_id"))
+                .withParentExternalId(rs.getString("parent_transaction_id"))
```
Comment (sfount) [inline src/main/java/uk/gov/pay/ledger/transaction/dao/mapper/TransactionMapper.java]:
```suggestion
                .withParentExternalId(rs.getString("parent_exernal_id"))
```

File: src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java [hunk 4]
```diff
@@ -29,19 +29,20 @@
...
                 "CAST(:externalMetadata as jsonb),:createdDate,CAST(:transactionDetails as jsonb), :eventCount," +
                 ":cardBrand,:lastDigitsCardNumber,:firstDigitsCardNumber,:netAmount,:totalAmount,:fee," +
                 ":transactionType::transaction_type" +
             ")" +
             "ON CONFLICT (external_id) " +
             "DO UPDATE SET " +
                 "external_id = EXCLUDED.external_id," +
+                "parent_transaction_id = EXCLUDED.parent_transaction_id," +
```
Comment (sfount) [inline src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java]:
```suggestion
                "parent_external_id = EXCLUDED.parent_external_id," +
```

File: src/main/java/uk/gov/pay/ledger/event/model/TransactionEntityFactory.java [hunk 5]
```diff
@@ -29,6 +29,7 @@ public TransactionEntity create(EventDigest eventDigest) {
         entity.setState(TransactionState.fromSalientEventType(eventDigest.getMostRecentEventType()).getState());
         entity.setCreatedDate(eventDigest.getEventCreatedDate());
         entity.setExternalId(eventDigest.getResourceExternalId());
+        entity.setParentExternalId(eventDigest.getParentResourceExternalId());
```
Comment (sfount) [inline src/main/java/uk/gov/pay/ledger/event/model/TransactionEntityFactory.java]: 👍

**Pull request merged**